### PR TITLE
Add a download script which downloads a lilac-processed dataset from HuggingFace

### PR DESCRIPTION
**Pull request by nsthorat**
Usage:

```
lilac download https://huggingface.co/datasets/lilacai/nikhil_staging-local-Capybara
```

You can pass optional args:
```
--project_dir: The project dir.
--dataset_namespace: Local namespace to write to.
--dataset_name: Local dataset name (to rewrite the dataset name).
--hf_token: A hf read token
```

https://huggingface.co/datasets/lilacai/nikhil_staging-local-Capybara

We also now upload a dataset-specific config when we upload datasets.

### CODE REVIEW

File: lilac/cli.py [hunk 1]
```diff
@@ -208,6 +209,44 @@ def deploy_project_command(
...
+  default='local',
+)
+@click.option(
+  '--dataset_name',
+  help='The local dataset name to use. Defaults to the name of the HuggingFace dataset.',
+  type=str,
+)
+@click.option(
+  '--hf_token',
+  help='The HuggingFace access token to use when making datasets private. '
```
Comment (brilee) [inline lilac/cli.py:235]: token to use when accessing private datasets?

**Review by dsmilkov — APPROVED** [review]

File: lilac/data/dataset_storage_utils.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  try:
+    from huggingface_hub import snapshot_download
+  except ImportError:
+    raise ImportError(
+      'Could not import the "huggingface_hub" python package. '
+      'Please install it with `pip install "huggingface_hub".'
+    )
+
+  if url_or_repo.startswith('https://huggingface.co/'):
+    repo_id = url_or_repo[len('https://huggingface.co/datasets/') :]
```
Comment (brilee) [inline lilac/data/dataset_storage_utils.py:48]: url_or_repo.removeprefix(...) is now a thing in python!

File: lilac/data/dataset_storage_utils.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  os.makedirs(dataset_dir, exist_ok=True)
+
+  shutil.copytree(src, dataset_dir, dirs_exist_ok=True)
+
+  ds = get_dataset(dataset_namespace, dataset_name)
+  dataset_config_filename = os.path.join(dataset_dir, 'dataset_config.yml')
+  if os.path.exists(dataset_config_filename):
+    with open(dataset_config_filename) as f:
+      dataset_config_yaml = f.read()
+    try:  # Try using the fast loader.
```
Comment (brilee) [inline lilac/data/dataset_storage_utils.py:88]: should abstract this try-except into a util?

**Review by brilee — APPROVED** [review]

File: lilac/data/dataset_storage_utils.py (hunk 2 above)
Comment (nsthorat) [reply lilac/data/dataset_storage_utils.py:48]: whoa

File: lilac/data/dataset_storage_utils.py (hunk 3 above)
Comment (nsthorat) [reply lilac/data/dataset_storage_utils.py:88]: done

**Pull request merged**